mikrotik-rs/src/protocol/word.rs

```diff
@@ -165,21 +164,39 @@ impl Display for WordCategory {
 pub struct WordAttribute<'a> {
     /// The key of the attribute.
     pub key: &'a str,
-    /// The value of the attribute, if present.
+    /// The value of the attribute, if present and in valid UTF-8.
     pub value: Option<&'a str>,
+    /// The value of the attribute, if present, in bytes.
+    pub value_raw: Option<&'a [u8]>,
 }
 
-impl<'a> TryFrom<&'a str> for WordAttribute<'a> {
+impl<'a> TryFrom<&'a [u8]> for WordAttribute<'a> {
     type Error = WordError;
 
-    fn try_from(value: &'a str) -> Result<Self, Self::Error> {
-        let mut parts = value
-            .strip_prefix('=')
-            .ok_or(WordError::Attribute)?
-            .splitn(2, '=');
-        let key = parts.next().ok_or(WordError::Attribute)?;
-        let value = parts.next();
-        Ok(Self { key, value })
+    fn try_from(word: &'a [u8]) -> Result<Self, Self::Error> {
+        // First byte must be '=' for attributes
+        if word.is_empty() || word[0] != b'=' {
+            return Err(WordError::Attribute);
+        }
+
+        // Find the second '=' that separates key from value
+        let mut parts = word[1..].splitn(2, |&b| b == b'=');
+
+        // Key part must exist and be valid UTF-8
+        let key_bytes = parts.next().ok_or(WordError::Attribute)?;
+        let key = std::str::from_utf8(key_bytes).map_err(|_| WordError::AttributeKeyNotUtf8)?;
```

Comment:
If this error happens, the method ends here. So the desired raw value is still not accessible...

Reply:
My assumption is that keys are always ASCII (so UTF-8 compatible) because are defined by RouterOS, while values can be whatever the user inputs. But I can be wrong on this. Can also keys be user-defined?

Reply:
You are right. My fault. Sorry.